northstar/src/npk/npk.rs

```diff
@@ -748,6 +758,92 @@ fn sign_hashes(key_pair: &Keypair, hashes_yaml: &str) -> String {
     )
 }
 
+fn set_selinux_contexts(root: &Path, context_type: &NonNullString) -> Result<(), Error> {
+    for dir_entry in WalkDir::new(root)
+        .follow_links(false)
+        .into_iter()
+        .filter_map(Result::ok)
+    {
+        let path = dir_entry.path();
+        if let Ok(path_c) = std::ffi::CString::new(path.as_os_str().as_bytes()) {
+            // Read existing selinux context
+            let mut context_buf = [0u8; 256];
```

Comment:
Why 256? Is this the max length of xattr values?

Reply:
We now use another call to getxattr to determine the proper size.